src/Evolution/Systems/GeneralizedHarmonic/Characteristics.cpp

```diff
@@ -100,8 +99,7 @@ void compute_characteristic_fields(
 }
 
 template <size_t Dim, typename Frame>
-typename Tags::CharacteristicFields<Dim, Frame>::type
-CharacteristicFieldsCompute<Dim, Frame>::function(
+typename Tags::CharacteristicFields<Dim, Frame>::type characteristic_fields(
```

Comment:
I agree @nilsdeppe! But should this be in the same PR as the upwind flux? The only reason I'm touching this file at all is to rename the functions, so I can call them in the upwind flux using the correct naming convention. 

Whether we do it in a separate PR or not, to confirm, you're suggesting these free functions should explicitly specify their types, instead of using Tags::...::type, right? If so, to make sure I understand, should line 102 above be the following? If not, what should it be?

~~~
Variables<tmpl::list<Tags::UPsi<Dim, Frame>, Tags::UZero<Dim, Frame>,
                                    Tags::UPlus<Dim, Frame>, Tags::UMinus<Dim, Frame>>> characteristic_fields(
~~~

Reply:
Yes, I'm am saying we should use the types explicitly instead of their tags. I'm fine doing this as either a separate commit in this PR or a separate PR (submitted in a timely manner so that we don't have people write new code base of this code and then have a snowball effect to deal with :) ).

Reply:
I’m happy to do this promptly, would prefer a separate PR that can go in quick. How do you avoid tags when using a Variables? Don’t you need tags for the keys?